database/memorydb.go

```diff
@@ -0,0 +1,81 @@
+package database
+
+import (
+	"fmt"
+	"sync"
+)
+
+type todos map[int]string
+
+type Database struct {
+	mu     sync.Mutex
+	data   todos
+	lastId int
+}
+
+func CreateDatabase() *Database {
+	return &Database{data: make(todos), lastId: -1}
+}
+
+func (db *Database) FetchTodo(id int) (string, error) {
+	var err error
+	db.mu.Lock()
+	defer db.mu.Unlock()
+
+	val, exists := db.data[id]
+
+	if !exists {
+		err = fmt.Errorf("ID: %v doesn't exist", id)
+	}
+
+	return val, err
+}
+
+func (db *Database) FetchAll() map[int]string {
+	db.mu.Lock()
```

Comment:
Same as above

Reply:
I think the same as the fetch one.
I this case it can be "worse" since I'm creating a map with a given length, so if it's not locked and there's an insertion or a removal it won't perform as good as it could